pkg/backporter/httpcache.go

```diff
@@ -0,0 +1,146 @@
+package backporter
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"net/http"
+	"net/http/httputil"
+	"sync"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	xFromCache       = "X-From-Cache"
+	cacheRefreshFreq = 10 * time.Minute
+)
+
+type bugzillaCache struct {
+	lock  sync.Mutex
+	cache map[string][]byte
+}
+
+func (bc *bugzillaCache) get(key string) ([]byte, bool) {
+	bc.lock.Lock()
+	defer bc.lock.Unlock()
+	cachedVal, ok := bc.cache[key]
+	return cachedVal, ok
+}
+
+func (bc *bugzillaCache) set(key string, respBytes []byte) {
+	bc.lock.Lock()
+	defer bc.lock.Unlock()
+	bc.cache[key] = respBytes
+}
+
+func (bc *bugzillaCache) delete(key string) {
+	bc.lock.Lock()
+	defer bc.lock.Unlock()
+	delete(bc.cache, key)
+}
+func newBugzillaCache() *bugzillaCache {
+	return &bugzillaCache{cache: map[string][]byte{}}
+}
+
+// cachingTransport is an implementation http.RoundTripper
+// which first checks for cached values
+type cachingTransport struct {
+	cache     *bugzillaCache
+	transport http.RoundTripper
+}
+
+// RoundTrip will first check if there are any cached responses and return that
+// if not it will make an HTTP call using the DefaultTransport
+func (t *cachingTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	if req.Method != "GET" {
+		return t.transport.RoundTrip(req)
+	}
+	var err error
+	var resp *http.Response
+	g := new(errgroup.Group)
+	g.Go(func() error {
+		resp, err = t.transport.RoundTrip(req)
+		if err != nil {
+			t.cache.delete(req.URL.String())
+			return err
+		}
+		body, err := httputil.DumpResponse(resp, true)
+		if err != nil {
+			t.cache.delete(req.URL.String())
+			return fmt.Errorf("err while prepping response to cache: %w", err)
+		}
+		if resp.StatusCode == http.StatusOK || resp.StatusCode == http.StatusNotFound {
+			t.cache.set(req.URL.String(), body)
+		} else {
+			t.cache.delete(req.URL.String())
+		}
+		return nil
+	})
+
+	if cachedVal, isCached := t.cache.get(req.URL.String()); isCached {
+		b := bytes.NewBuffer(cachedVal)
+		cachedResp, err := http.ReadResponse(bufio.NewReader(b), req)
+		if err != nil {
+			return nil, err
+		}
+		cachedResp.Header.Set(xFromCache, "1")
+		return cachedResp, nil
+	}
+	if err := g.Wait(); err != nil {
+		return nil, err
+	}
+	return resp, nil
+}
+
+func refreshCache(bc *bugzillaCache) {
+	concurrencyLimiter := make(chan string, 10)
```

Comment:
Would be nice to log here that we are doing our periodic refresh and how many bugs we have in the cache. The latter might also might sense as a metric

Reply:
I have added logging and metrics here

Reply:
Would you mind to use a field for this:
`logrus.WithField("cache_entries", len(bc.cache)).Info("Refreshing cache")`

This has the advantage that we can easely filter for this in the logs later on, because we can filter for the `"Refreshing cache"` message, right now we would need a regex